openverse_catalog/dags/maintenance/rotate_db_snapshots.py

```diff
@@ -0,0 +1,91 @@
+"""
+Manages weekly database snapshots. RDS does not support weekly snapshots
+schedules on its own, so we need a DAG to manage this for us.
+
+It runs on Saturdays at 00:00 UTC in order to happen before the data refresh.
+
+The DAG will automatically delete the oldest snapshots when more snaphots
+exist than it is configured to retain.
+
+Requires two variables:
+
+`AIRFLOW_RDS_ARN`: The ARN of the RDS DB instance that needs snapshots.
+`AIRFLOW_RDS_SNAPSHOTS_TO_RETAIN`: How many historical snapshots to retain.
+"""
+
+import logging
+from datetime import datetime
+
+import boto3
+from airflow.decorators import dag, task
+from airflow.models import Variable
+from airflow.providers.amazon.aws.operators.rds import RdsCreateDbSnapshotOperator
+from airflow.providers.amazon.aws.sensors.rds import RdsSnapshotExistenceSensor
+
+
+logger = logging.getLogger(__name__)
+
+DAG_ID = "check_silenced_dags"
+MAX_ACTIVE = 1
+
+AIRFLOW_RDS_ARN = Variable.get("AIRFLOW_RDS_ARN")
+AIRFLOW_RDS_SNAPSHOTS_TO_RETAIN = Variable.get(
+    "AIRFLOW_RDS_SNAPSHOTS_TO_RETAIN", default_var="7"
+)
```

Comment:
As a side note, we can also use Variable's [`deserialize_json` parameter](https://airflow.apache.org/docs/apache-airflow/stable/concepts/variables.html) to convert this to an `int` implicitly. This would become:

```suggestion
AIRFLOW_RDS_SNAPSHOTS_TO_RETAIN = Variable.get(
    "AIRFLOW_RDS_SNAPSHOTS_TO_RETAIN",
    default_var=7,
    deserialize_json=True,
)
```
In this case if the value in the Variable field the string `"14"`, `AIRFLOW_RDS_SNAPSHOTS_TO_RETAIN` would then become `14`.